.orchestra/ci/ci-hook/app.py

```diff
@@ -89,77 +159,309 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         assert not self.user.impersonationtokens.get(impersonation_token_id).active
 
 
-def trigger_ci(username, repo_url, ref, before, after):
+def impersonate(admin_gl, username):
+    matching = admin_gl.users.list(username=username)
+    if len(matching) != 1:
+        raise Exception("Unexpected number of users matching " + username)
+    user_id = matching[0].id
+
+    user = admin_gl.users.get(user_id)
+    return ImpersonationToken(user)
+
+
+def trigger_ci(username, repo_url, base_repo_url, ref, status_update_metadata: Optional[dict] = None):
     # Ignore pushes by the CI itself
     if username == ci_user:
         return
 
     admin_gl = gitlab.Gitlab(GITLAB_URL, private_token=ADMIN_TOKEN)
     admin_gl.auth()
 
-    matching = admin_gl.users.list(username=username)
-    if len(matching) != 1:
-        raise Exception("Unexpected number of users matching " + username)
-    user_id = matching[0].id
+    try:
+        with impersonate(admin_gl, username) as token:
+            user_gl = gitlab.Gitlab(GITLAB_URL, private_token=token)
+            user_gl.auth()
+            project = user_gl.projects.get(PROJECT_ID)
+
+            is_anonymous = username == default_user
+
+            variables = {
+                "TARGET_COMPONENTS_URL": repo_url,
+                "TARGET_COMPONENTS": " ".join(is_anonymous and default_user_target_components or target_components),
+                "PUSHED_REF": ref,
+                "ORCHESTRA_CONFIG_REPO_HTTP_URL": ORCHESTRA_CONFIG_REPO_HTTP_URL,
+                "ORCHESTRA_CONFIG_REPO_SSH_URL": ORCHESTRA_CONFIG_REPO_SSH_URL,
+            }
+
+            if status_update_metadata is not None:
+                variables["REVNG_CI_STATUS_UPDATE_METADATA"] = json.dumps(status_update_metadata)
```

Comment:
1. Does this contain any sensitive information?
2. This does not affect the behavior of the job, it's just metadata for later consumption. Is this right?

Add some docs.

Reply:
1. No
2. Yes, it only contains data required to submit back the job status to the pull request. If not set, no status updates are sent.

Reply:
Gotcha, add some docs.